python-checks/src/main/java/org/sonar/python/checks/cdk/IamPolicyPublicAccessCheck.java

```diff
@@ -0,0 +1,172 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks.cdk;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import javax.annotation.Nullable;
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.IssueLocation;
+import org.sonar.plugins.python.api.SubscriptionContext;
+import org.sonar.plugins.python.api.tree.CallExpression;
+import org.sonar.plugins.python.api.tree.DictionaryLiteral;
+import org.sonar.plugins.python.api.tree.Expression;
+import org.sonar.plugins.python.api.tree.ListLiteral;
+import org.sonar.plugins.python.api.tree.RegularArgument;
+import org.sonar.python.tree.TreeUtils;
+
+import static org.sonar.python.checks.cdk.CdkPredicate.isFqn;
+import static org.sonar.python.checks.cdk.CdkPredicate.isFqnOf;
+import static org.sonar.python.checks.cdk.CdkPredicate.isString;
+import static org.sonar.python.checks.cdk.CdkUtils.getArgument;
+import static org.sonar.python.checks.cdk.CdkUtils.getCall;
+import static org.sonar.python.checks.cdk.CdkUtils.getDictionary;
+import static org.sonar.python.checks.cdk.CdkUtils.getDictionaryPair;
+import static org.sonar.python.checks.cdk.CdkUtils.getList;
+import static org.sonar.python.checks.cdk.CdkUtils.getListElements;
+import static org.sonar.python.checks.cdk.CdkUtils.getListExpression;
+import static org.sonar.python.checks.cdk.CdkUtils.resolveDictionary;
+
+@Rule(key = "S6270")
+public class IamPolicyPublicAccessCheck extends AbstractCdkResourceCheck {
+
+  private static final String ISSUE_MESSAGE = "Make sure granting public access is safe here.";
+  private static final String SECONDARY_MESSAGE = "Related effect.";
+
+  private static final String IAM_EFFECT_ALLOW = "aws_cdk.aws_iam.Effect.ALLOW";
+
+  @Override
+  protected void registerFqnConsumer() {
+    checkFqn("aws_cdk.aws_iam.PolicyStatement", IamPolicyPublicAccessCheck::checkPolicyStatementConstructor);
+    checkFqn("aws_cdk.aws_iam.PolicyStatement.from_json", IamPolicyPublicAccessCheck::checkPolicyStatementJson);
+    checkFqn("aws_cdk.aws_iam.PolicyDocument.from_json", IamPolicyPublicAccessCheck::checkPolicyDocumentJson);
+  }
+
+  private static void checkPolicyStatementConstructor(SubscriptionContext subscriptionContext, CallExpression callExpression) {
+    // The rule should only trigger if "effect" is set to "ALLOW" or if it is unset
+    Optional<CdkUtils.ExpressionFlow> effect = getArgument(subscriptionContext, callExpression, "effect");
+    getArgument(subscriptionContext, callExpression, "principals")
+      .flatMap(CdkUtils::getListExpression)
+      .ifPresent(principals -> {
+        Optional<Expression> allowSetting = effect.flatMap(expressionFlow -> expressionFlow.getExpression(isFqn(IAM_EFFECT_ALLOW)));
+        if (effect.isEmpty() || allowSetting.isPresent()) {
+          checkPrincipals(subscriptionContext, allowSetting.orElse(null), principals);
```

Comment:
```suggestion
           getListElements(ctx, principals)
      .forEach(expressionFlow -> raiseIssueIf(expressionFlow, isSensitivePrincipal(), effect))
```
isn't `IamPolicyPublicAccessCheck#raiseIssueIf` doing the same job as `checkPrincipals`